lib/src/db/models.dart

```diff
@@ -53,8 +53,8 @@ class Key {
     return obj as Partition;
   }
 
-  Key append(Type modelType, {Object id}) {
-    return new Key(this, modelType, id);
+  Key<T> append<T>(Type modelType, {T id}) {
```

Comment:
This should be rather with a different type, so that parent types can have a different id type:
````
Key<R> append<R>(Type modelType, {R id}) {
````

Reply:
Good catch :)